src/main/scala/chisel3/ltl/LTL.scala

```diff
@@ -408,7 +408,7 @@ sealed abstract class AssertPropertyLike(defaultLayer: Layer) {
     *   emitted as `label: assert(...)`.
     */
   def apply(
-    prop:    Property,
+    prop:    => Property,
```

Comment:
Interesting fixup. LGTM

Reply:
Why is this change here? If you look below, `prop` is accessed multiple times so now this is going to execute multiple times.

Reply:
It is intended to delay `prop` to be evaluated in the `block`. Actually, it will only execute once since the definition of `fold` is
```scala
@inline final def fold[B](ifEmpty: => B)(f: A => B): B =
  if (isEmpty) ifEmpty else f(this.get)
```

Reply:
Good point and very clever, perhaps too clever 🙂. This was not obvious to me (despite the fact that I should know better), so I suspect it won't be obvious to others in the future updating this code. So in the interest of defensive programming against future bugs, please create a val above `val clocked` (but obviously inside the `block` that you assign `prop` to with a comment saying "evaluate by name parameter once", and then use that value in the `fold`.

Reply:
Thank you for the suggestion.